lib/web/connection_diagnostic.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"net/http"
+
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/web/ui"
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+)
+
+// getConnectionDiagnostic returns a connection diagnostic connection diagnostics.
+func (h *Handler) getConnectionDiagnostic(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	connectionID := p.ByName("connectionid")
+	connectionDiagnostic, err := clt.GetConnectionDiagnostic(r.Context(), connectionID)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	resLabels := connectionDiagnostic.GetAllLabels()
+	uiLabels := make([]ui.Label, len(resLabels))
+	for labelName, labelValue := range resLabels {
+		uiLabels = append(uiLabels, ui.Label{
+			Name:  labelName,
+			Value: labelValue,
+		})
+	}
```

Comment:
What does connection test result need labels for?

Reply:
It doesn't.
I added them because I thought that it was a common pattern.
Removing it :+1: